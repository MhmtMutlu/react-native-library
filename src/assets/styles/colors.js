const colors = {
  grey: '#6c757d',
  darkGrey: '#495057',
  darkGrey2x: '#343a40',
  lightGrey: '#adb5bd',
  lightGrey2x: '#dee2e6',
  lightGrey3x: '#e0e0e0',
  white: '#f8f9fa',
};

export default colors;
