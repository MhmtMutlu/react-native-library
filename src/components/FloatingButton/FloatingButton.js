import React from 'react';
import {TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import colors from '../../assets/styles/colors';
import styles from './FloatingButton.syles';

const FloatingButton = ({onPress, icon}) => {
  return (
    <TouchableOpacity
      testID="floating-button"
      style={styles.container}
      onPress={onPress}>
      <Icon name={icon} color={colors.darkGrey2x} size={35} />
    </TouchableOpacity>
  );
};

export default FloatingButton;
