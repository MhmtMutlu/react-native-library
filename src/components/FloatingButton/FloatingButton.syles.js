import {StyleSheet} from 'react-native';
import colors from '../../assets/styles/colors';

export default StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 20,
    right: 20,
    borderRadius: 50,
    width: 60,
    height: 60,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.white,
    shadowColor: 'black',
    shadowOpacity: 0.5,
    elevation: 8,
    shadowRadius: 20,
    shadowOffset: {width: 56, height: 10},
  },
});
