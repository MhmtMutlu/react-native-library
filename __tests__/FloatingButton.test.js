import React from 'react';
import {render} from '@testing-library/react-native';
import FloatingButton from '../src/components/FloatingButton';
import styles from '../src/components/FloatingButton/FloatingButton.syles';

test('should match with snapshot', () => {
  const comp = render(<FloatingButton />);
  expect(comp).toMatchSnapshot();
});

test('should render icon correctly', () => {
  const testText = 'plus';
  const comp = render(<FloatingButton icon={testText} />);
  const floatingButton =
    comp.getByTestId('floating-button').children[0]._fiber.stateNode.props.name;
  expect(floatingButton).toBe(testText);
});

test('should render FloatingButton css', () => {
  const comp = render(<FloatingButton />);
  const floatingButton = comp.getByTestId('floating-button').props.style;
  expect(floatingButton).toMatchSnapshot(styles.container);
});
