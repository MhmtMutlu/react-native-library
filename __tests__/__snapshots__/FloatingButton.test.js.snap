// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f8f9fa",
      "borderRadius": 50,
      "bottom": 20,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "right": 20,
      "width": 60,
    }
  }
  testID="floating-button"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#343a40",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`should render FloatingButton css 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#f8f9fa",
  "borderRadius": 50,
  "bottom": 20,
  "height": 60,
  "justifyContent": "center",
  "opacity": 1,
  "position": "absolute",
  "right": 20,
  "width": 60,
}
`;
